File: app/src/main/java/com/unis/gameplatfrom/model/GameModel.java
package com.unis.gameplatfrom.model;

import com.unis.gameplatfrom.api.PublicApiInterface;
import com.unis.gameplatfrom.api.RetrofitWrapper;
import com.unis.gameplatfrom.api.result.BaseCustomListResult;
import com.unis.gameplatfrom.constant.GameConstant;
import com.unis.gameplatfrom.entity.GamesEntity;

import io.reactivex.Observable;

public class GameModel implements GameConstant.Model {

    @Override
    public Observable<BaseCustomListResult<GamesEntity>> getGameList(String uuid) {
        return RetrofitWrapper.getInstance().create(PublicApiInterface.class).getGameList(uuid);
    }
}

File: kotlin/src/main/java/com/unis/kotlin/model/GameModel.kt
package com.unis.kotlin.model

import com.unis.kotlin.api.PublicApiInterface
import com.unis.kotlin.api.RetrofitWrapper
import com.unis.kotlin.api.result.BaseCustomListResult
import com.unis.kotlin.constant.GameConstant
import com.unis.kotlin.entity.GamesEntity
import io.reactivex.Observable

class GameModel : GameConstant.Model{


    override fun getGameList(uuid: String): Observable<BaseCustomListResult<GamesEntity>> {
        return RetrofitWrapper.get().create(PublicApiInterface::class.java)
                .getGameList(uuid)
    }


}